app/services/ListRetrievalService.scala

```diff
@@ -32,8 +32,30 @@ class ListRetrievalService @Inject() (listRepository: ListRepository)(implicit e
   private def getVersion = MetaData("version", LocalDate.of(2020, 11, 5))
 
   def getList(listName: ListName): Future[Option[ReferenceDataList]] =
-    listRepository.getList(listName, getVersion).map {
-      x => Some(ReferenceDataList(listName, getVersion, x))
+    listRepository.getListByName(listName, getVersion).map {
+      x =>
+        Some(ReferenceDataList(listName, getVersion, x))
     }
 
+  def getResourceLinks(metaData: Option[MetaData] = None): Future[Option[ResourceLinks]] =
+    listRepository.getAllLists.map {
+      list =>
+        if (list.nonEmpty)
+          Some(ResourceLinks(_links = Some(buildLinks(list)), metaData = metaData))
+        else None
+    }
+
+  private def buildLinks(list: List[GenericListItem]): Map[String, JsObject] = {
+
+    val buildUri: String => String =
+      uri => s"/customs-reference-data/$uri"
+
+    val resourceLinks: Seq[Map[String, JsObject]] = list.zipWithIndex.map {
+      case (data, index) => Map(s"list${index + 1}" -> JsObject(Seq("href" -> JsString(buildUri(data.listName.listName)))))
+    }
+
+    Map("self" -> JsObject(Seq("href" -> JsString(buildUri("lists"))))) ++
+      resourceLinks.flatten
+  }
```

Comment:
Should this be on the constructor of `ResourceLinks` or it's apply method?

Reply:
That's a good shout. That way if we needed to have a different base uri for another set of links (not sure why) it would be tied into the relevant ResourceLinks implementation.